O/opam/build_tarballs.jl

```diff
@@ -0,0 +1,33 @@
+using BinaryBuilder, Pkg
+
+name = "opam"
+version = v"2.3.0"
+
+sources = [
+    ArchiveSource("https://github.com/ocaml/opam/releases/download/2.3.0/opam-full-2.3.0.tar.gz",
+                  "506ba76865dc315b67df9aa89e7abd5c1a897a7f0a92d7b2694974fdc532b346")
+]
+
+script = raw"""
+cd $WORKSPACE/srcdir/opam-full-2.3.0
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --with-vendored-deps
+make -j
+make install
+install_license LICENSE
+"""
+
+platforms = [
+    Platform("x86_64", "linux"; libc="glibc"),
+    Platform("x86_64", "linux"; libc="musl")
+]
+
+products = [
+    ExecutableProduct("opam", :opam),
+]
+
+dependencies = [
+    BuildDependency(PackageSpec(name="OCaml_jll", version=v"5.3.0"))
```

Comment:
Leaving a note here that the reason this was required to be a `BuildDependency` not `HostBuildDependency`, because our OCaml packages hardcode the library search path to `/destdir`.

Reply:
`OCAMLLIB` is apparently a workaround for that: https://discuss.ocaml.org/t/stdlib-directory-relative-to-the-compiler-bins/2940

Reply:
We can also patch OCaml to add a relative path option.

Reply:
https://github.com/JuliaPackaging/Yggdrasil/pull/10389